eclair-core/src/test/scala/fr/acinq/eclair/integration/IntegrationSpec.scala

```diff
@@ -260,6 +262,85 @@ class IntegrationSpec extends TestKit(ActorSystem("test")) with BitcoindService
     awaitAnnouncements(nodes.filterKeys(key => !List("A", "B").contains(key)), 10, 12, 24)
   }
 
+  test("open a wumbo channel and wait for longer than the default min_depth") {
+    // we open a 5BTC channel and check that we scale `min_depth` up to 13 confirmations
+    val funder = nodes("C")
+    val fundee = nodes("F1")
+    val sender = TestProbe()
+    sender.send(funder.switchboard, Peer.OpenChannel(
+      remoteNodeId = fundee.nodeParams.nodeId,
+      fundingSatoshis = 5.btc,
+      pushMsat = 0.msat,
+      fundingTxFeeratePerKw_opt = None,
+      channelFlags = None,
+      timeout_opt = None))
+    val tempChannelId = sender.expectMsgType[ChannelOpened](10 seconds).channelId
+
+    // mine the funding tx
+    generateBlocks(bitcoincli, 2)
+
+    // get the channelId
+    sender.send(fundee.register, 'channels)
+    val Some((_, fundeeChannel)) = sender.expectMsgType[Map[ByteVector32, ActorRef]].find(_._1 == tempChannelId)
+    sender.send(fundeeChannel, CMD_GETSTATEDATA)
+    val channelId = sender.expectMsgType[HasCommitments].channelId
+
+    awaitCond({
+      sender.send(funder.register, Forward(channelId, CMD_GETSTATE))
+      sender.expectMsgType[State] == WAIT_FOR_FUNDING_LOCKED
+    })
+
+    generateBlocks(bitcoincli, 6)
+
+    // after 8 blocks the fundee is still waiting for more confirmations
+    sender.send(fundee.register, Forward(channelId, CMD_GETSTATE))
+    assert(sender.expectMsgType[State] == WAIT_FOR_FUNDING_CONFIRMED)
+
+    // after 8 blocks the funder is still waiting for funding_locked from the fundee
+    sender.send(funder.register, Forward(channelId, CMD_GETSTATE))
+    assert(sender.expectMsgType[State] == WAIT_FOR_FUNDING_LOCKED)
+
+    // simulate a disconnection
+    sender.send(funder.switchboard, Peer.Disconnect(fundee.nodeParams.nodeId))
+    assert(sender.expectMsgType[String] == "disconnecting")
+
+    awaitCond({
+      sender.send(fundee.register, Forward(channelId, CMD_GETSTATE))
+      Try(sender.expectMsgType[State]) == Success(OFFLINE)
```

Comment:
Wasn't that the real reason of the random failure in the tests?
Because you would start trying to reconnect while fundee may have not been offline yet?
If that's the case it seems you won't need the `awaitCond` below, it's surprising that we need it if peers are both correctly offline when we send the reconnect.

Reply:
Right the result of the first check was thrown away and the `Try` were redundant, i force-pushed a fix. Yes the reason it failed is that we begun the reconnection attempt while the peer wasn't offline yet, note that we check the channel STATE not the peer STATE and the 2 peers have more than one channel between them. So after the first `Peer.Connect` was dismissed with `"already connected"` we didn't try again. The `awaitCond` is handy to repeat the connection attempt, note that there might be one superfluous attempt due to the transition OFFLINE -> SYNCING -> WAIT_FOR_FUNDING_CONFIRMED; the alternative is to wait until the `Peer` is in state OFFLINE but feels a bit cumbersome.

Reply:
FYI [this](https://github.com/araspitzu/eclair/commit/d6c371feb15bd497417802739a0c91e3ad8dd220) is the test with peer disconnection, we still have an `awaitCond` on the peer STATE :/